gr-blocks/lib/file_source_impl.cc

```diff
@@ -100,33 +126,61 @@ namespace gr {
 
       // we use "open" to use to the O_LARGEFILE flag
       if((fd = ::open(filename, O_RDONLY | OUR_O_LARGEFILE | OUR_O_BINARY)) < 0) {
-	perror(filename);
-	throw std::runtime_error("can't open file");
+        perror(filename);
```

Comment:
nope, let's not do `perror` if we could also `GR_LOG_ERROR(d_logger, boost::format("file_source(%s): %s") % filename % strerror(errno))`, or so.